File: pubnub-gson/src/main/java/com/pubnub/api/models/consumer/PNPage.java
package com.pubnub.api.models.consumer;

import lombok.AllArgsConstructor;
import lombok.Getter;

@AllArgsConstructor
public abstract class PNPage {
    @Getter
    protected final String hash;

    public static Next next(String hash) {
        return new Next(hash);
    }

    public static Previous previous(String hash) {
        return new Previous(hash);
    }

    public static class Next extends PNPage {
        Next(String hash) {
            super(hash);
        }
    }

    public static class Previous extends PNPage {
        Previous(String hash) {
            super(hash);
        }
    }
}

File: pubnub-entities/src/main/kotlin/com/pubnub/entities/objects/PNPage.kt
package com.pubnub.entities.objects

sealed class PNPage : com.pubnub.api.models.consumer.PNPage {
    internal abstract val pageHash: String
    data class PNNext(override val pageHash: String) : PNPage()
    data class PNPrev(override val pageHash: String) : PNPage()
}
